Content.Shared/Guidebook/GuidebookData.cs

```diff
@@ -0,0 +1,62 @@
+using Robust.Shared.Serialization;
+using Robust.Shared.Utility;
+
+namespace Content.Shared.Guidebook;
+
+/// <summary>
+/// Used by GuidebookDataSystem to hold data extracted from prototype values,
+/// both for storage and for network transmission.
+/// </summary>
+[Serializable, NetSerializable]
+[DataDefinition]
+public sealed partial class GuidebookData
+{
+    /// <summary>
+    /// Total number of data values stored.
+    /// </summary>
+    [DataField]
+    public int Count { get; private set; }
+
+    /// <summary>
+    /// The data extracted by the system.
+    /// </summary>
+    /// <remarks>
+    /// Structured as PrototypeName, ComponentName, FieldName, Value
+    /// </remarks>
+    [DataField]
+    public Dictionary<string, Dictionary<string, Dictionary<string, object?>>> Data = [];
+
```

Comment:
You're better off turning this into a FrozenDictionary as they're faster and better if you build the data once / rarely but access it frequently (it's something like 100x slower to build but like 2x faster to access).